### PR TITLE
chore: consolidate security groups to save eni usage

### PR DESCRIPTION
**Pull request by amliuyong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [x] all code changes are covered by unit tests
- [x] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [x] with KDS sink
    - [ ] with S3 sink
  - [x] deploy data processing
  - [ ] deploy data modeling
    - [x] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/ingestion-server/server/private/sg.ts [hunk 1]
```diff
@@ -69,24 +71,8 @@ export function createALBSecurityGroup(
...
-    },
-    {
-      id: 'W5',
-      reason: 'Design intent: Security Groups found with cidr open to world on egress',
-    },
-  ]);
+): ISecurityGroup {
 
-  return ec2Sg;
+  return createSGForEgressToAwsService(scope, `${RESOURCE_ID_PREFIX}ecs-sg`, vpc);
```
Comment (zxkane) [inline src/ingestion-server/server/private/sg.ts:76]: rename `ecs-sg` to `ECSEgressToAWSServiceSG`

Comment (zxkane) [inline src/ingestion-server/server/private/sg.ts:76]: rename `ecs-sg` to `ECSEgressToAWSServiceSG`

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/analytics/analytics-on-redshift.ts [hunk 2]
```diff
@@ -90,6 +91,8 @@ export class RedshiftAnalyticsStack extends NestedStack {
 
     this.templateOptions.description = `(${SolutionInfo.SOLUTION_ID}-dmr) ${SolutionInfo.SOLUTION_NAME} - ${featureName} ${SolutionInfo.SOLUTION_VERSION_DETAIL}`;
 
+    const securityGroupForLambda = createSGForEgressToAwsService(this, 'FnSg', props.vpc);
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply src/analytics/analytics-on-redshift.ts]: changed.

File: src/ingestion-server/kinesis-data-stream/private/sg.ts [hunk 3]
```diff
@@ -11,19 +11,13 @@
...
-): SecurityGroup {
-  const sg = new SecurityGroup(scope, 'lambdaKinesisToS3Sg', {
-    description: 'Security group for kinesis to s3 lambda',
-    vpc,
-    allowAllOutbound: true,
-  });
-  addCfnNagToSecurityGroup(sg, ['W40', 'W5']);
-  return sg;
+): ISecurityGroup {
+  return createSGForEgressToAwsService(scope, 'FnSg', vpc);
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply src/ingestion-server/kinesis-data-stream/private/sg.ts]: changed.

File: src/ingestion-server/server/private/sg.ts (hunk 1 above)
Comment (amliuyong) [reply src/ingestion-server/server/private/sg.ts]: changed

**Review by zxkane — APPROVED** [review]

**Pull request merged**